# VacationPy
----

#### Create a heat map that displays the humidity for every city from WeatherPy


In [79]:
# Dependencies and Setup
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import requests
import time
import json
from scipy.stats import linregress
from pprint import pprint
import country_converter as coco
from scipy import stats
import gmaps

# Import API key
from api_keys import weather_api_key
from config import gkey

# Incorporated citipy to determine city based on latitude and longitude
from citipy import citipy

# configure Gmaps
gmaps.configure(api_key=gkey)
fig = gmaps.figure()

# Output File (CSV)
output_data_file = "vacation_output_data/cities.csv"

# Range of latitudes and longitudes
lat_range = (-90, 90)
lng_range = (-180, 180)

## Re-Generate Cities List

In [2]:
# List for holding lat_lngs and cities
lat_lngs = []
cities = []

# Create a set of random lat and lng combinations
lats = np.random.uniform(lat_range[0], lat_range[1], size=1500)
lngs = np.random.uniform(lng_range[0], lng_range[1], size=1500)
lat_lngs = zip(lats, lngs)

# Identify nearest city for each lat, lng combination
for lat_lng in lat_lngs:
    city = citipy.nearest_city(lat_lng[0], lat_lng[1]).city_name
    
    # If the city is unique, then add it to a our cities list
    if city not in cities:
        cities.append(city)

# Print the city count to confirm sufficient count
len(cities)

643

### Perform API Calls
* Perform a weather check on each city using a series of successive API calls.
* Include a print log of each city as it'sbeing processed (with the city number and city name).


In [3]:
# Build the query for the for-loop
url = "http://api.openweathermap.org/data/2.5/weather?"
units = "imperial"
query_url = f"{url}appid={weather_api_key}&units={units}&q="  



In [4]:
# set up lists to hold reponse info
lat = []
long = []
temp = []
humid = []
clouds = []
winds = []
city1 = []
city2 = []
max_temp = []
country = []
datetime = []
print_request = []
print_response = []
print_dne = []

In [5]:
# Loops work on smaller range.  Reset and empty list by running codeblock where they created.
# Run loops and monitor print outputs.

In [6]:
for city in range(len(cities)):
    try:
        weather2 = requests.get(query_url + str(cities[city])).json()
        lat.append(weather2['coord']['lat'])
        long.append(weather2['coord']['lon'])
        temp.append(weather2['main']['feels_like'])
        max_temp.append(weather2['main']['temp_max'])
        humid.append(weather2['main']['humidity'])
        clouds.append(weather2['clouds']['all'])
        winds.append(weather2['wind']['speed'])
        city1.append(weather2['name'])
        country.append(weather2['sys']['country'])
        datetime.append(weather2['dt'])
        print(f"Request {city} sent for {cities[city]}.")
        print_request.append(f"Request {city} sent for {cities[city]}.")
        print(f"Response code {weather2['cod']} received for {cities[city]}.")
        print_response.append(f"Response code {weather2['cod']} received for {cities[city]}.")
    except:
        print(f"{cities[city]} does not exist in OpenWeatherMap.")
        city2.append(cities[city])
        print_dne.append(f"{cities[city]} does not exist in OpenWeatherMap.")
        pass

    

Request 0 sent for fatehpur.
Response code 200 received for fatehpur.
Request 1 sent for jamestown.
Response code 200 received for jamestown.
Request 2 sent for ushuaia.
Response code 200 received for ushuaia.
belushya guba does not exist in OpenWeatherMap.
Request 4 sent for castro.
Response code 200 received for castro.
Request 5 sent for chokurdakh.
Response code 200 received for chokurdakh.
Request 6 sent for quatre cocos.
Response code 200 received for quatre cocos.
Request 7 sent for butaritari.
Response code 200 received for butaritari.
Request 8 sent for athabasca.
Response code 200 received for athabasca.
Request 9 sent for rikitea.
Response code 200 received for rikitea.
Request 10 sent for saint-francois.
Response code 200 received for saint-francois.
Request 11 sent for busselton.
Response code 200 received for busselton.
Request 12 sent for spassk-ryazanskiy.
Response code 200 received for spassk-ryazanskiy.
Request 13 sent for punta arenas.
Response code 200 received for 

Request 116 sent for chulumani.
Response code 200 received for chulumani.
Request 117 sent for araguaina.
Response code 200 received for araguaina.
Request 118 sent for sinnamary.
Response code 200 received for sinnamary.
armacao dos buzios does not exist in OpenWeatherMap.
Request 120 sent for hasaki.
Response code 200 received for hasaki.
Request 121 sent for pevek.
Response code 200 received for pevek.
Request 122 sent for saint anthony.
Response code 200 received for saint anthony.
katsiveli does not exist in OpenWeatherMap.
Request 124 sent for kodiak.
Response code 200 received for kodiak.
Request 125 sent for punta alta.
Response code 200 received for punta alta.
Request 126 sent for furano.
Response code 200 received for furano.
Request 127 sent for east london.
Response code 200 received for east london.
Request 128 sent for deputatskiy.
Response code 200 received for deputatskiy.
Request 129 sent for marzuq.
Response code 200 received for marzuq.
Request 130 sent for ewa beac

Request 231 sent for zhangjiakou.
Response code 200 received for zhangjiakou.
Request 232 sent for ayan.
Response code 200 received for ayan.
Request 233 sent for atlantic beach.
Response code 200 received for atlantic beach.
Request 234 sent for klaksvik.
Response code 200 received for klaksvik.
Request 235 sent for lakes entrance.
Response code 200 received for lakes entrance.
Request 236 sent for chuy.
Response code 200 received for chuy.
halalo does not exist in OpenWeatherMap.
Request 238 sent for buta.
Response code 200 received for buta.
Request 239 sent for tiksi.
Response code 200 received for tiksi.
Request 240 sent for hibbing.
Response code 200 received for hibbing.
Request 241 sent for ambilobe.
Response code 200 received for ambilobe.
Request 242 sent for melendugno.
Response code 200 received for melendugno.
Request 243 sent for saint-philippe.
Response code 200 received for saint-philippe.
Request 244 sent for sao filipe.
Response code 200 received for sao filipe.
Reque

Request 351 sent for cabedelo.
Response code 200 received for cabedelo.
Request 352 sent for gorin.
Response code 200 received for gorin.
Request 353 sent for dangriga.
Response code 200 received for dangriga.
Request 354 sent for epping.
Response code 200 received for epping.
jabiru does not exist in OpenWeatherMap.
Request 356 sent for bow island.
Response code 200 received for bow island.
Request 357 sent for virginia beach.
Response code 200 received for virginia beach.
Request 358 sent for alofi.
Response code 200 received for alofi.
Request 359 sent for itoman.
Response code 200 received for itoman.
Request 360 sent for pangai.
Response code 200 received for pangai.
Request 361 sent for godfrey.
Response code 200 received for godfrey.
wahran does not exist in OpenWeatherMap.
Request 363 sent for malindi.
Response code 200 received for malindi.
Request 364 sent for barranca.
Response code 200 received for barranca.
Request 365 sent for chenzhou.
Response code 200 received for chen

Request 469 sent for banda aceh.
Response code 200 received for banda aceh.
Request 470 sent for lata.
Response code 200 received for lata.
Request 471 sent for truth or consequences.
Response code 200 received for truth or consequences.
Request 472 sent for novoagansk.
Response code 200 received for novoagansk.
Request 473 sent for sur.
Response code 200 received for sur.
Request 474 sent for moundou.
Response code 200 received for moundou.
falealupo does not exist in OpenWeatherMap.
Request 476 sent for emerald.
Response code 200 received for emerald.
inderborskiy does not exist in OpenWeatherMap.
Request 478 sent for tianpeng.
Response code 200 received for tianpeng.
Request 479 sent for jieshi.
Response code 200 received for jieshi.
Request 480 sent for clyde river.
Response code 200 received for clyde river.
Request 481 sent for bonavista.
Response code 200 received for bonavista.
Request 482 sent for aksu.
Response code 200 received for aksu.
Request 483 sent for cognac.
Response

Request 588 sent for takoradi.
Response code 200 received for takoradi.
Request 589 sent for wageningen.
Response code 200 received for wageningen.
Request 590 sent for san pedro carcha.
Response code 200 received for san pedro carcha.
Request 591 sent for denpasar.
Response code 200 received for denpasar.
Request 592 sent for okha.
Response code 200 received for okha.
Request 593 sent for airai.
Response code 200 received for airai.
Request 594 sent for otjimbingwe.
Response code 200 received for otjimbingwe.
Request 595 sent for atar.
Response code 200 received for atar.
Request 596 sent for tazovskiy.
Response code 200 received for tazovskiy.
hihifo does not exist in OpenWeatherMap.
Request 598 sent for sinegorye.
Response code 200 received for sinegorye.
Request 599 sent for urumqi.
Response code 200 received for urumqi.
Request 600 sent for maceio.
Response code 200 received for maceio.
Request 601 sent for guatire.
Response code 200 received for guatire.
Request 602 sent for semi

### Convert Raw Data to DataFrame
* Export the city data into a .csv.
* Display the DataFrame

In [7]:
city_weather_dict ={"city": city1, "latitude": lat, "longitude": long, "Max Temp": max_temp,
                   "humidity": humid, "Feels Like": temp, "cloudiness": clouds, "wind speed": winds,
                   "country": country, "datetime": datetime}

In [8]:
city_weather = pd.DataFrame(city_weather_dict)
city_weather.head()

city  latitude  longitude  Max Temp  humidity  Feels Like  \
0    Fatehpur   25.9333    80.8000     77.68        91       79.43   
1   Jamestown   42.0970   -79.2353     74.84        68       74.95   
2     Ushuaia  -54.8000   -68.3000     47.86        66       45.25   
3      Castro  -24.7911   -50.0119     58.15        80       57.43   
4  Chokurdakh   70.6333   147.9167     56.95        83       56.25   

   cloudiness  wind speed country    datetime  
0          99        9.08      IN  1661110697  
1          75        5.75      US  1661110697  
2         100        5.75      AR  1661110698  
3          97        4.90      BR  1661110699  
4         100        3.74      RU  1661110699

In [9]:
# unix, UTC timestamp conversion from:
# https://www.codegrepper.com/code-examples/python/convert+unix+timestamp+to+datetime+python+pandas
city_weather["date"] = pd.to_datetime(city_weather["datetime"], unit="s").dt.tz_localize("US/Pacific")

city_weather.head()

city  latitude  longitude  Max Temp  humidity  Feels Like  \
0    Fatehpur   25.9333    80.8000     77.68        91       79.43   
1   Jamestown   42.0970   -79.2353     74.84        68       74.95   
2     Ushuaia  -54.8000   -68.3000     47.86        66       45.25   
3      Castro  -24.7911   -50.0119     58.15        80       57.43   
4  Chokurdakh   70.6333   147.9167     56.95        83       56.25   

   cloudiness  wind speed country    datetime                      date  
0          99        9.08      IN  1661110697 2022-08-21 19:38:17-07:00  
1          75        5.75      US  1661110697 2022-08-21 19:38:17-07:00  
2         100        5.75      AR  1661110698 2022-08-21 19:38:18-07:00  
3          97        4.90      BR  1661110699 2022-08-21 19:38:19-07:00  
4         100        3.74      RU  1661110699 2022-08-21 19:38:19-07:00

In [10]:
# found country converter to convert from ISO 2 letter country to country name at the following"
# https://pypi.org/project/country-converter/
# https://stackoverflow.com/questions/64051741/how-to-convert-country-code-to-name-with-country-converter-in-pandas
# https://notebook.community/konstantinstadler/country_converter/doc/country_converter_examples
converter = coco.CountryConverter()
city_weather["country name"] = converter.convert(city_weather["country"], src="ISO2", to="name_short")

In [11]:
city_weather.head()

city  latitude  longitude  Max Temp  humidity  Feels Like  \
0    Fatehpur   25.9333    80.8000     77.68        91       79.43   
1   Jamestown   42.0970   -79.2353     74.84        68       74.95   
2     Ushuaia  -54.8000   -68.3000     47.86        66       45.25   
3      Castro  -24.7911   -50.0119     58.15        80       57.43   
4  Chokurdakh   70.6333   147.9167     56.95        83       56.25   

   cloudiness  wind speed country    datetime                      date  \
0          99        9.08      IN  1661110697 2022-08-21 19:38:17-07:00   
1          75        5.75      US  1661110697 2022-08-21 19:38:17-07:00   
2         100        5.75      AR  1661110698 2022-08-21 19:38:18-07:00   
3          97        4.90      BR  1661110699 2022-08-21 19:38:19-07:00   
4         100        3.74      RU  1661110699 2022-08-21 19:38:19-07:00   

    country name  
0          India  
1  United States  
2      Argentina  
3         Brazil  
4         Russia

In [12]:
city_weather2 = city_weather[["city", "latitude", "longitude", "Max Temp", "humidity",
                             "Feels Like", "cloudiness", "wind speed", "country name",
                             "date"]] 
city_weather2.head()

city  latitude  longitude  Max Temp  humidity  Feels Like  \
0    Fatehpur   25.9333    80.8000     77.68        91       79.43   
1   Jamestown   42.0970   -79.2353     74.84        68       74.95   
2     Ushuaia  -54.8000   -68.3000     47.86        66       45.25   
3      Castro  -24.7911   -50.0119     58.15        80       57.43   
4  Chokurdakh   70.6333   147.9167     56.95        83       56.25   

   cloudiness  wind speed   country name                      date  
0          99        9.08          India 2022-08-21 19:38:17-07:00  
1          75        5.75  United States 2022-08-21 19:38:17-07:00  
2         100        5.75      Argentina 2022-08-21 19:38:18-07:00  
3          97        4.90         Brazil 2022-08-21 19:38:19-07:00  
4         100        3.74         Russia 2022-08-21 19:38:19-07:00

In [13]:
city_weather2.to_csv("vacation_output_data/cities.csv")
len(city_weather2)


586

In [14]:
request_response_dict = {"city": city1, "request sent": print_request, "response received": print_response}
request_response = pd.DataFrame(request_response_dict)
request_response.head()

city                    request sent  \
0    Fatehpur    Request 0 sent for fatehpur.   
1   Jamestown   Request 1 sent for jamestown.   
2     Ushuaia     Request 2 sent for ushuaia.   
3      Castro      Request 4 sent for castro.   
4  Chokurdakh  Request 5 sent for chokurdakh.   

                            response received  
0    Response code 200 received for fatehpur.  
1   Response code 200 received for jamestown.  
2     Response code 200 received for ushuaia.  
3      Response code 200 received for castro.  
4  Response code 200 received for chokurdakh.

In [15]:
request_response.to_csv("vacation_output_data/request_response_log.csv", index=False)
len(request_response)

586

In [16]:
city_dne_dict = {"city": city2, "response received": print_dne}
city_dne = pd.DataFrame(city_dne_dict)
city_dne.head()

city                                  response received
0     belushya guba    belushya guba does not exist in OpenWeatherMap.
1  illoqqortoormiut  illoqqortoormiut does not exist in OpenWeather...
2         taolanaro        taolanaro does not exist in OpenWeatherMap.
3       nizhneyansk      nizhneyansk does not exist in OpenWeatherMap.
4         umzimvubu        umzimvubu does not exist in OpenWeatherMap.

In [17]:
city_dne.to_csv("vacation_output_data/response_error_log.csv", index=False)

len(city_dne)

57

## Narrow down the DataFrame to find your ideal weather condition. For example:
- A max temperature lower than 80 degrees but higher than 70.
- Wind speed less than 10 mph.
- Zero cloudiness.
- Drop any rows that don't satisfy all three conditions. You want to be sure the weather is ideal.
----
- **Note:** Feel free to adjust your specifications, but make sure to limit the number of rows returned by your API requests to a reasonable number.
- Try to reduce the results in your DataFrame to 10 or fewer cities.

In [214]:
# Using Feels Like temperature since it considers humidity and temperature together in a Heat Index.
# Running this .loc with Feels Like > 70 degrees and < 8 degrees.
# With latitude between -30 degrees South and 30 degrees North.  "Some place tropical."
# With cloudiness < 25% and winds less than 10  mph.
# Adjust humidity percent until get to 10 or less results.
ideal_feels = city_weather2.loc[(city_weather2["Feels Like"] > 70) & (city_weather2["Feels Like"] < 80), :]
#ideal_lat = ideal_feels.loc[(ideal_feels["latitude"] > -25) & (ideal_feels["latitude"] < 25), :]                             
ideal_clouds = ideal_feels.loc[(ideal_feels["cloudiness"] == 0), :]
ideal_winds = ideal_clouds.loc[(ideal_clouds["wind speed"] < 10), :]
#ideal_humidity = ideal_winds.loc[(ideal_winds["humidity"] < 70), :]
#ideal_feels
#len(ideal_feels) # 177
#ideal_lat
#len(ideal_lat) # 111
#ideal_clouds
#len(ideal_clouds) # 22
#ideal_winds
#len(ideal_winds) # 13
#ideal_humidity = ideal_feels.loc[(ideal_feels["humidity"] < 70), :]
#ideal_humidity
#ideal_lat = ideal_humidity.loc[(ideal_humidity["latitude"] > -30) & (ideal_feels["latitude"] < 30), :]
#ideal_clouds = ideal_lat.loc[(ideal_lat["cloudiness"] < 75), :]
#ideal_clouds
#len(ideal_lat) # 4
#ideal_winds
#len(ideal_winds) # 10
ideal = ideal_winds.reset_index(drop=True)
ideal


city  latitude  longitude  Max Temp  humidity  \
0                      Nantucket   41.2835   -70.0995     80.71        60   
1                        Terrace   54.5163  -128.6035     76.91        43   
2                        Hibbing   47.4272   -92.9377     76.57        38   
3                Havre-St-Pierre   50.2334   -63.5986     77.36        61   
4  Libertador General San Martín  -23.8064   -64.7876     75.97        15   
5                     Vermillion   42.7794   -96.9292     79.81        38   
6                        Tacuatí  -23.4500   -56.5833     77.50        49   
7                           Jadu   31.9530    12.0261     73.49        55   
8                        Korčula   42.9622    17.1369     77.09        67   
9                         Wajima   37.4000   136.9000     75.70        79   

   Feels Like  cloudiness  wind speed   country name                      date  
0       72.05           0        6.91  United States 2022-08-21 19:37:06-07:00  
1       76.32           0        4.61         Canada 2022-08-21 19:39:04-07:00  
2       73.44           0        6.91  United States 2022-08-21 19:40:22-07:00  
3       77.67           0        4.61         Canada 2022-08-21 19:41:04-07:00  
4       73.98           0        4.88      Argentina 2022-08-21 19:41:14-07:00  
5       78.96           0        4.41  United States 2022-08-21 19:41:39-07:00  
6       77.25           0        4.07       Paraguay 2022-08-21 19:41:54-07:00  
7       73.13           0        8.55          Libya 2022-08-21 19:42:24-07:00  
8       77.65           0        9.78        Croatia 2022-08-21 19:42:25-07:00  
9       76.69           0        8.57          Japan 2022-08-21 19:42:34-07:00

## Plotting the Data
* Use proper labeling of the plots using plot titles (including date of analysis) and axes labels.
* You must take a screenshot of the heatmap that you create and include it in your submission
* Google Maps
------
* Use Google Places API to find the first hotel for each city located within 5,000 meters of your coordinates.
* Plot the hotels on top of the humidity heatmap, with each pin containing the **Hotel Name**, **City**, and **Country**.
* A heatmap is successfully created.
* For max intensity in the heatmap, try setting it to the highest humidity found in the dataset.
* Ten or less pins for all the cities in the DataFrame.
* Pins are clickable to display City, Country and Hotel Name.
* Placed on top of the heatmap.

## Humidity Heatmap

In [20]:
# Use Activities 3 Airport Ratings for loops and radius
# Use Activities 3 airport_heatmap for heatmap
# Use Activites 3 Banking_Desert_HeatMap for heatmap markers


In [261]:
# Store latitude and longitude in locations
locations = ideal_winds[["latitude", "longitude"]]
len(locations)

10

In [216]:
#places = [ideal_winds[["latitude", "longitude"]]]
#print(places)

In [217]:
# type(locations)  # DataFrame

In [259]:
humid = ideal_winds["humidity"]
#humid
#type(humid)
len(humid)

10

In [270]:
# Customize the size of the map
figure_layout = {
    'width': '80%',
    'margin': '0 auto 0 auto',
    'padding': '1px',
    'border': '1px solid black'
}

# Plot Heatmap
fig = gmaps.figure(layout=figure_layout, map_type='SATELLITE',
                  zoom_level=2, center=(30.267, -97.743),
                  display_toolbar=False)

# I TURNED OFF display_toolbar IN ORDER TO DOWNLOAD A PNG OF THE HEATMAP.  I TURNED IT BACK ON BEFORE I FILED.

# Create heatmap layer
heatmap_layer = gmaps.heatmap_layer(locations, weights=humid,
                                dissipating=False, max_intensity=79,
                                point_radius=5, opacity = 1.0)

# Add layer
fig.add_layer(heatmap_layer)


# Display figure
fig

Figure(layout=FigureLayout(border='1px solid black', height='420px', margin='0 auto 0 auto', padding='1px', wi…

### If I say lived in a cruddy climate, where it was windy and rainy all of the time, and I wanted to travel somewhere with ideal weather right now, I might define the location based on ideal weather conditions consisting of cloudless days, winds less than 10 mph, and the temperature is between 70 and 80 degrees Fahrenheit.  Someone wants to make my dream come true, but I can only choose from a list of 500 - 600 randomly selected cities anywhere in the world before I apply my weather conditions.  As it turns out, I only get to choose from the 10 results shown on the heatmap, and the list started with 586 cities! 


### This heatmap is based on humidity with the city in Japan having the highest intensity, and the city in Argentina having the lowest intensity. 

## Humidity Heatmap with Markers

In [307]:
# Store city and country name in location_name
location_name = ideal_winds[["city", "country name"]]
#type(location_name)
#print(location_name)

In [200]:
# From Activities 3 Google Places
# From Activities 3 NearestRestr
# From Activities 3 Airport_Ratings

In [303]:
# set up additional columns to hold results from requests
#places = locations[["latitude", "longitude"]]
#places.loc[index, "name"] = ""
#places.loc[index, "address"] = ""
#places.loc[index, "rating"] = ""
#places.head()

In [304]:
# geocoordinates
# use locations list
#target_search = "hotel"
#target_radius = 5000
#target_type = "lodging"
#target_ranking = "rankby"

#            # set up a parameters dictionary
#
#            #    "keyword": "hotel",
#            params = {
#                "radius": 1000,
#                "type": "lodging",
#                "key": gkey
#            }
#            #    "keyword": target_search,
#
#
#
#            # use iterrows to iterate through the locations DataFrame
#            for index, row in places.iterrows():
#                # get lat, lng from df
#                lat = row["latitude"]
#                lng = row["longitude"]
#
#                # change location each iteration while leaving original params in place
#                params["location"] = f"{lat},{lng}"
#
#                # Use the search term: "hotel" and our lat/lng
#                base_url = "https://maps.googleapis.com/maps/api/place/nearbysearch/json"
#
#                # base url and API requests
#                print(f"Retrieving Results for Index {index}: {locations}.")
#                hotel_name = requests.get(base_url, params=params)
#
#                # find lodging within 5000 meters of the city center
#                # extract results
#                hotel_name = hotel_name.json()
#
#                # use try-except to clear any exceptions
#                try:
#                    #print(f"Lodging available is {hotel_name["results"][0]["name"]}.")
#
#                    places.loc[index, "name"] = hotel_name["results"][0]["name"]
#                    places.loc[index, "address"] = hotel_name["results"][0]["vicinity"]
#                    #places.loc[index, "price_level"] = hotel_name["results"][0]["price_level"]
#                    places.loc[index, "rating"] = hotel_name["results"][0]["rating"]
#
#                except (KeyError, IndexError):
#                    print("Missing field/result... skipping.")
#
#                print("------------")
#
#
#            
# NEED MORE TIME TO GET THIS CODE BLOCK TO WORK.  LARGE INDENTS ARE FOR THE CODE THAT NEEDS TO BE ADJUSTED.

In [191]:
# check response url 
# !!!!!!!COMMENT OUT THIS PRINT STATEMENT SO OUTPUT WITH THE API KEY IS NOT EXPOSED WHEN COMMIT TO GITHUB!!!!
#print(response.url)

In [190]:
# Converting response to json
# places_data = response.json()

In [305]:
# Print the json to see results
#print(json.dumps(hotel_name, indent=4, sort_keys=True))
#pprint(hotel_name)
# THIS WORKS, BUT I ONLY GET THE RESULTS FOR THE LAST CITY LOCATION.  NEED TO ADJUST LOOP IN CODE BLOCK ABOVE.

In [306]:
test_city = ideal_winds["city"].tolist()
test_country = ideal_winds["country name"].tolist()
city_country = list(zip(test_city, test_country))
#print(city_country)


In [308]:
# Customize the size of the map
figure_layout = {
    'width': '80%',
    'margin': '0 auto 0 auto',
    'padding': '1px',
    'border': '1px solid black'
}

# Plot Heatmap
fig = gmaps.figure(layout=figure_layout, map_type='SATELLITE',
                  zoom_level=2, center=(30.267, -97.743),
                  display_toolbar=False)
# I TURNED OFF display_toolbar IN ORDER TO DOWNLOAD A PNG OF THE HEATMAP.  I TURNED IT BACK ON BEFORE I FILED.

# Create heatmap layer
heatmap_layer = gmaps.heatmap_layer(locations, weights=humid,
                                dissipating=False, max_intensity=79,
                                point_radius=5, opacity = 1.0)

# Assign the marker layer to markers
markers = gmaps.marker_layer(locations,
                            display_info_box=True,
                            info_box_content=test_city,
                            hover_text=test_city)
                             
#                            hover_text=['Nantucket', 'Terrace', 'Hibbing', 'Havre-St-Pierre', 'Libertador General San Martín', 'Vermillion',
#         'Tacuatí', 'Jadu', 'Korčula', 'Wajima'])

#hover_text="bluh\nbluh\nbluh")

# Add the layer to the map
fig.add_layer(markers)



# Add layer
fig.add_layer(heatmap_layer)


# Display figure
fig

Figure(layout=FigureLayout(border='1px solid black', height='420px', margin='0 auto 0 auto', padding='1px', wi…

### Well, if I would at least want to know if there is a hotel in each of these cities.

### Markers are clickable, but they only have the city name right now.  Hover_text also works, but only with city name so far.
-------

### I need to spend more time on the code block for the Google Places nearbyseach API results to get the markers to fully function with city, country, hotel.